packages/techdocs-common/src/stages/publish/awsS3.ts

```diff
@@ -61,9 +62,30 @@ export class AwsS3Publish implements PublisherBase {
     );
     let accessKeyId = undefined;
     let secretAccessKey = undefined;
+    let awsCredentials:
+      | Credentials
+      | CredentialsOptions
+      | undefined = undefined;
     if (credentials) {
-      accessKeyId = credentials.getOptionalString('accessKeyId');
-      secretAccessKey = credentials.getOptionalString('secretAccessKey');
+      const roleArn = credentials.getOptionalString('roleArn');
+      if (roleArn && aws.config.credentials instanceof Credentials) {
+        awsCredentials = new aws.ChainableTemporaryCredentials({
+          masterCredentials: aws.config.credentials as Credentials,
+          params: {
+            RoleSessionName: 'backstage-aws-organization-processor',
```

Comment:
Can this be anything?

Reply:
yeah, pretty much. Usually you want it to be something descriptive since the session name shows up in AWS CloudTrail audit logs

Reply:
Cool! :)